hir-analysis/src/validation/naming.rs

```diff
@@ -0,0 +1,272 @@
+use miden_diagnostics::{DiagnosticsHandler, Severity, SourceSpan, Spanned};
+use miden_hir::*;
+
+use super::{Rule, ValidationError};
+
+/// This validation rule ensures that all identifiers adhere to the rules of their respective items.
+pub struct NamingConventions;
+/// Module naming conventions:
+///
+/// 1. Must not be empty
+/// 2. Must begin with a lowercase ASCII alphabetic character
+/// 3. May otherwise consist of any number of characters of the following classes:
+///   * `A-Z`
+///   * `a-z`
+///   * `0-9`
+///   * `_-+$@`
+/// 4. May only contain `:` when used via the namespacing operator, e.g. `std::math`
+/// 5. The namespacing operator may only appear between two valid module identifiers
+/// 6. Namespaced module names must adhere to the above rules in each submodule identifier
+impl Rule<Module> for NamingConventions {
+    fn validate(
+        &mut self,
+        item: &Module,
```

Comment:
When I see `item` in the body I need to translate it to the `module`. Would it be better to `item` -> `module`? OTOH, I see the style in having the validation subject named the same way in every rule, but in this case, I'd lean on the side of readability.

Reply:
Yeah I was shooting for consistency, but it's an easy change to make, so I'm +1